R/template.R

```diff
@@ -77,6 +73,244 @@ metric_summarizer <- function(metric_nm,
+    finalize_estimator(!! truth, estimator, metric_nm)
+  )
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = metric_nm,
+    .estimator = eval_tidy(finalize_estimator_expr),
+    .estimate = metric_fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      !!! spliceable_estimator(estimator),
+      na_rm = na_rm,
+      !!! spliceable_event_level(event_level),
+      !!! spliceable_case_weights(case_weights),
+      !!! metric_fn_options
+    )
+  )
+
+  dplyr::as_tibble(metric_tbl)
+}
+
+#' Developer function for summarizing new numeric metrics
+#'
+#' `metric_summarizer_numeric()` is useful alongside [metric_vec_template()] for
+#' implementing new custom metrics. `metric_summarizer_numeric()` calls the
+#' metric function inside `dplyr::summarise()`. `metric_vec_template()` is a
+#' generalized function that calls the core implementation of a metric function,
+#' and includes a number of checks on the types, lengths, and argument inputs.
+#' See [Custom performance
+#' metrics](https://www.tidymodels.org/learn/develop/metrics/) for more
+#' information.
+#'
+#' @details
+#'
+#' `metric_summarizer_numeric()` is generally called from the data frame version
+#' of your metric function. It knows how to call your metric over grouped data
+#' frames and returns a `tibble` consistent with other metrics.
+#'
+#'
+#' @inheritParams metric_summarizer
+#' @param metric_nm A single character representing the name of the metric to
+#' use in the `tibble` output.
+#'
+#' @param estimate Generally, the unquoted column name corresponding to
+#' the `estimate` column.
+#'
+#' @seealso [metric_vec_template()] [finalize_estimator()] [dots_to_estimate()]
+#'   [metric_summarizer_class()] [metric_summarizer_prob()]
+#'
+#' @export
+metric_summarizer_numeric <- function(metric_nm,
+                                      metric_fn,
+                                      data,
+                                      truth,
+                                      estimate,
+                                      na_rm = TRUE,
+                                      case_weights = NULL,
+                                      ...,
+                                      metric_fn_options = list()) {
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+  case_weights <- enquo(case_weights)
+
+  validate_not_missing(truth, "truth")
+  validate_not_missing(estimate, "estimate")
+
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  finalize_estimator_expr <- rlang::expr(
+    finalize_estimator(!! truth, metric_class = metric_nm)
+  )
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = metric_nm,
+    .estimator = eval_tidy(finalize_estimator_expr),
+    .estimate = metric_fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      na_rm = na_rm,
+      !!! spliceable_case_weights(case_weights),
```

Comment:
I've got some ideas to make `truth`, `estimate`, `case_weights`, and `...` use tidyselect rather than allowing arbitrary expressions through here (which in retrospect is a bit horrifying). We can talk about it after this first round of review, but I think it will simplify these functions a bit further.

If I'm right, we will be able to do something like `.data[[truth]]` rather than `!!` which would be nice

Reply:
I think that will also automatically support this special case so we would be able to remove

```r
  # Explicit handling of length 1 character vectors as column names
  nms <- colnames(data)
  truth <- handle_chr_names(truth, nms)
  estimate <- handle_chr_names(estimate, nms)
```

That would work automatically because tidyselection allows strings as well as bare expressions